drivers/i2s/i2s_ll_stm32.c

```diff
@@ -837,37 +857,46 @@ struct queue_item rx_1_ring_buf[CONFIG_I2S_STM32_RX_BLOCK_COUNT + 1];
 struct queue_item tx_1_ring_buf[CONFIG_I2S_STM32_TX_BLOCK_COUNT + 1];
 
 static struct i2s_stm32_data i2s_stm32_data_1 = {
-	.dma_name = I2S1_DMA_NAME,
 	.rx = {
-		.dma_channel = I2S1_DMA_CHAN_RX,
+		.dma_name = DT_I2S_1_DMA_CONTROLLER_RX,
+		.dma_channel = DT_I2S_1_DMA_CHANNEL_RX,
 		.dma_cfg = {
 			.block_count = 1,
-			.dma_slot = I2S1_DMA_SLOT_RX,
+			.dma_slot = DT_I2S_1_DMA_SLOT_RX,
 			.channel_direction = PERIPHERAL_TO_MEMORY,
 			.source_data_size = 1,  /* 16bit default */
 			.dest_data_size = 1,    /* 16bit default */
```

Comment:
I think that .source_data_size and .dest_data_size should be 2 (16bit == 2 bytes)